### ISSUE TITLE
SVG tag parsing

### ISSUE THREAD

**Issue by rbfrank** [gutenbergtools/ebookmaker#275]
Ebookmaker apparently does not accept svg files created with `tex2svg`.  These generated tags result in CRITICAL errors: role, Focusable, aria-labelledby, and aria-hidden.

Should ebookmaker be parsing SVG image files? And if it does, is there a description of tags it will or will not allow? From that list, it seems that aria-hidden should be allowed to control exposure to screen readers.

**Comment by eshellman** [comment]
Focusable is not included in SVG 1.1, which is used by EPUB.
If you get the other error (making sure that the file is parsed as html, not xml) please post the sample file generating the errors and reopen the issue.

**Comment by rbfrank** [comment]
[altitude2.zip](https://github.com/user-attachments/files/20445326/altitude2.zip)

[output.txt](https://github.com/user-attachments/files/20445327/output.txt)

**Comment by eshellman** [comment]
that is html flavored SVG - put it in the html file and it works fine.

**Comment by rbfrank** [comment]
I'm confused. The zip file provided above has an HTML file that uses that SVG. Is that what you mean by "put it in the html file" or is there some other way other than as a normal illustration? And does "it works fine" mean I can ignore the CRITICAL errors in output.txt?

**Comment by eshellman** [comment]
embed the html code in the html document.

**Comment by rtonsing** [comment]
Now that I looked more into SVG, I understand that it is XML code, and _could_ be edited.

But I don't think it is reasonable to expect an average user to edit a file generated by an imaging application, much less pull out code to put in the HTML. Which code? All of it? Any but the most simple images quickly become gibberish to any but an expert, I certainly am not comfortable with it.

If it is necessary to parse & check SVG code for epub, we need to understand why, and have some guidelines on which editors generate good code. Users should not be expected to manually edit the files, any more than JPG or PNG files.

My 2c.

**Comment by rbfrank** [comment]
The tex2svg npm package is supposed to default to 1.1. It is configurable as well. But on further investigation, it's considered a relatively inactive project. Maybe we shouldn't be hitching our SVG wagon to that tool anyway. I don't know of anything better, though.

We, or more specifically DP, need a "HowTo" for making usable SVG 1.1 files from LaTeX source.

**Comment by jhellingman** [comment]
I ran into the same issue, and with chatGTP hacked together a little XSLT transformation to remove the offending attributes, as well as a perl script to apply it to a whole folder of svg files.

`<?xml version="1.0" encoding="UTF-8"?>
<xsl:stylesheet version="3.0"
  xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
  xmlns:svg="http://www.w3.org/2000/svg">

  
  <xsl:template match="@*|node()">
    <xsl:copy>
      <xsl:apply-templates select="@*|node()"/>
    </xsl:copy>
  </xsl:template>

  
  <xsl:template match="@*[starts-with(name(), 'aria-')]"/>

  
  <xsl:template match="svg:svg/@role"/>
  <xsl:template match="svg:svg/@focusable"/>

</xsl:stylesheet>`

And 

`#!/usr/bin/perl
use strict;
use warnings;
use File::Copy;

my $saxonHome = $ENV{'SAXON_HOME'};
my $javaOptions = '-Xms2048m -Xmx4096m -Xss1024k ';

my $java      = "java $javaOptions";
my $saxon     = "$java -jar " . $saxonHome . "/saxon9he.jar ";

# XSLT file (adjust if needed)
my $xslt_file = 'cleansvg.xsl';

# Check if the XSLT file exists
die "XSLT file '$xslt_file' not found\n" unless -e $xslt_file;

# Process each .svg file in the current directory
foreach my $svg_file (glob("*.svg")) {
    my $backup_file = "$svg_file.bak";
    my $temp_output = "$svg_file.tmp";

    print "Processing $svg_file -> (backup: $backup_file)\n";

    # Run the transformation
    my $cmd = "$saxon -s:$svg_file -xsl:$xslt_file -o:$temp_output";
    system($cmd) == 0 or do {
        warn "Failed to process $svg_file: $!\n";
        next;
    };

    # Rename original to .bak
    move($svg_file, $backup_file) or do {
        warn "Failed to backup $svg_file to $backup_file: $!\n";
        next;
    };

    # Move cleaned output to original name
    move($temp_output, $svg_file) or do {
        warn "Failed to write cleaned file to $svg_file: $!\n";
        # Restore from backup in case of error
        move($backup_file, $svg_file);
    };
}
`
